File: Fe-vue/src/entities/Document.ts
export interface Document {
  Id: string | null
  Name: string
  Path?: string
  FolderPath: string
  CreatedAt: string | null
  ParentId?: string | null
  CategoryId?: string | null
  Category?: DocumentCategory
  Type: DocumentType
  Children?: Document[]
  IsLoaded?: boolean
  IsExpend?: boolean
  MarkdownContent?: string

  DocumentBlocks: DocumentBlock[]
  Issuer: string | null
  IssueDate: string | null
  DocumentNo: string | null
  DocumentName: string | null
  SignerName: string | null
  EffectiveDate: string | null
  IsLaw: boolean
}

export interface DocumentCategory {
  Id: string
  Name: string
}

export enum DocumentType {
  Folder = 'Folder',
  Word = 'Word',
  Pdf = 'Pdf',
  Excel = 'Excel',
  Ppt = 'Ppt',
  Image = 'Image',
  Unknown = 'Unknown',
  Link = 'Link',
}

export interface DocumentBlock {
  Id: string | null
  ParentId: string | null
  DocumentId: string
  Title: string
  Content: string
  Level: number
  ContentType: number
  Order: number
  IsExpand?: boolean
  IsShow?: boolean
  State?: number /*0: khong thay doi, 1: cap nhat, 2: them moi, 3: xoa bo, 4: xoa sau khi them */
}

File: Back-End/Cukcuk.Core/Entities/Document.cs
using Cukcuk.Core.Enum;
using System.Text.Json.Serialization;

namespace Cukcuk.Core.Entities
{
    public class Document
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Path { get; set; } = string.Empty;
        public DateTime? CreatedAt { get; set; }
        public DocumentType Type { get; set; }
        public string FolderPath { get; set; } = string.Empty;

        public Guid? ParentId { get; set; }
        [JsonIgnore]
        public Document? Parent { get; set; }

        [JsonIgnore]
        public List<Document> Children { get; set; } = new List<Document>();

        public Guid? CategoryId { get; set; }
        public DocumentCategory? Category { get; set; }

        public List<DocumentBlock> DocumentBlocks { get; set; } = new List<DocumentBlock>();

        public string? Issuer { get; set; }

        public DateTime? IssueDate { get; set; }

        public string? DocumentNo { get; set; }

        public string? SignerName { get; set; }

        public bool IsLaw { get; set; } = false;

    }
}
